### PR TITLE
Migrating from checkstyle to spotless for formatting

### PR DESCRIPTION
**Pull request by VachaShah**
### Description
Adding spotless support in this repo in place of checkstyle so that we can get features for checking and applying the formatting using spotless. The rules used here are from the OpenSearch repo. spotlessJavaCheck is now part of the build task making it easier for developers to format their changes.

This PR adds the spotless functionality, makes it part of the build and enforces spotless on 3 modules - json, transport and util along with the samples folder. The commits are divided accordingly.

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by VachaShah** [comment]
@reta @dblock The next PRs will enable spotless for other modules. LMK what you think of this division!

File: java-client/src/main/java/org/opensearch/client/json/UnionDeserializer.java [hunk 1]
```diff
@@ -213,9 +218,9 @@ public JsonpDeserializer<Union> build() {
                 otherMembers.put(Event.START_OBJECT, objectMembers.remove(0));
             }
 
-//            if (objectMembers.size() > 1) {
-//                System.out.println("multiple objects in " + buildFn);
-//            }
+            // if (objectMembers.size() > 1) {
```
Comment (reta) [inline java-client/src/main/java/org/opensearch/client/json/UnionDeserializer.java:221]: Could we clean up such comments? :-)

File: java-client/src/main/java/org/opensearch/client/json/ObjectDeserializer.java [hunk 2]
```diff
@@ -280,33 +271,33 @@ public void setTypeProperty(String name, String defaultType) {
...
-//        });
-//    }
-//
-//    public static class JsonpIntParser {
-//        public int parse(JsonParser parser) {
-//            JsonpUtils.expectNextEvent(parser, Event.VALUE_NUMBER);
-//            return parser.getInt();
-//        }
-//    }
+    // Experiment: avoid boxing, allow multiple primitive parsers (e.g. int as number & string)
```
Comment (reta) [inline java-client/src/main/java/org/opensearch/client/json/ObjectDeserializer.java:281]: Could we clean up such comments? :-)

**Review by reta — APPROVED** [review]

File: java-client/src/main/java/org/opensearch/client/json/UnionDeserializer.java (hunk 1 above)
Comment (VachaShah) [reply java-client/src/main/java/org/opensearch/client/json/UnionDeserializer.java]: Done :)

File: java-client/src/main/java/org/opensearch/client/json/ObjectDeserializer.java (hunk 2 above)
Comment (VachaShah) [reply java-client/src/main/java/org/opensearch/client/json/ObjectDeserializer.java]: Done :)

**Review by reta — APPROVED** [review]

**Review by dblock — APPROVED** [review]

**Pull request merged**